Funcy.Test/ApplicativeLawsCheck.fs

```diff
@@ -164,3 +164,80 @@ module ApplicativeLawsCheck =
             apply ``Homomorphism in Either<TLeft, TRight>``
             apply ``Interchange in Either<TLeft, TRight>``
         }
+
+    module ApplicativeLawsInFuncyList =
+        let pureFList x = FuncyList.Construct([|x|])
+        
+        let ``Identity in Cons<T>`` = Prop.forAll(Arb.int)(fun i ->
+            let v = FuncyList.Cons(i, FuncyList.Nil())
+            // pure id <*> v = v
+            v.Apply(pureFList funcId) = (v :> FuncyList<int>)
+        )
+
+        let ``Identity in Nil<T>`` = Prop.forAll(Arb.int)(fun i ->
+            let v = FuncyList<int>.Nil()
+            // pure id <*> v = v
+            v.Apply(pureFList funcId) = (v :> FuncyList<int>)
+        )
+
+        let ``Composition in FuncyList<T> 1`` = Prop.forAll(Arb.systemFunc(CoArbitrary.int, Arb.int), Arb.systemFunc(CoArbitrary.int, Arb.int), Arb.int)(fun f g i ->
+            let u = FuncyList.Cons(f, FuncyList.Nil())
+            let v = FuncyList.Construct([|g|])
```

Comment:
These are to include diversity of types.
Cons returns Cons type value, Nil returns Nil type value, and Construct returns FuncyList type value.
I want to check the Functor laws with various types. :smile: